doc/design/draft-metadata-migration.md

```diff
@@ -0,0 +1,13 @@
+Premise: Draft metadata should always be in the latest schema version.
+
+There are a few places in the API where draft metadata are altered (POST, PUT and served (GET). Note that the UI and 
+API use a shared schema version. All actions should thus assume that this is the only version available to the UI and
+server. Any other versions need to be migrated to this version for any display or editing.
+
+1. The API should call dandischema `migrate` on all metadata exchanges of draft metadata.
+2. The API can have the option (parameter `?migrate=true`) of providing migration on a `GET` on published metadata. 
+   This would be needed for the UI to display landing pages for published dandisets with older schemas.
+3. To determine the state of validity, the API should also revalidate all draft metadata on any schema version change 
+   in the infrastructure.
```

Comment:
Right now the validation task uses the `schemaVersion` specified in the metadata to validate. Revalidation is pointless unless the `schemaVersion` is changed in some way. Do you mean that all draft metadata should be migrated to the newest version whenever the version is bumped? Or should the validation task be using the server's `schemaVersion`, regardless of what `schemaVersion` is specified in the metadata?

Reply:
> Do you mean that all draft metadata should be migrated to the newest version whenever the version is bumped?

correct.

Reply:
note: With my [proposed above](https://github.com/dandi/dandi-api/pull/475/files#r688588814) `/migrate` endpoint `?migrate=true` would not be strictly necessary. Any client (e.g. web ui) could explicitly inform user that it is migrating the metadata from X to Y, list warnings etc, and proceed with the migrated version.  If we do `?migrate=true` client would still ideally need to get first without that flag to inform user that version is outdated, and then re-query with `?migrate=true` but then response will be just metadata record, without any fields to carry back information about migration procedure, warnings etc.

Reply:
remember `?migrate=true` only applies to published dandisets (all draft dandisets auto migrate).

Reply:
then my `edit` in above [comment](https://github.com/dandi/dandi-api/pull/475/files#r688588814) was correct that `/migrate` should be at the level of versioned dandiset endpoint.